compiler/rustc_middle/src/middle/privacy.rs

```diff
@@ -27,26 +27,106 @@ pub enum AccessLevel {
     Public,
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug, HashStable, Default)]
+pub struct EffectiveVisibility {
+    public: Option<Visibility>,
+    exported: Option<Visibility>,
+    reachable: Option<Visibility>,
+    reachable_from_impl_trait: Option<Visibility>,
+}
+
+impl EffectiveVisibility {
+    pub fn get(&self, tag: AccessLevel) -> Option<&Visibility> {
+        match tag {
+            AccessLevel::Public => self.public.as_ref(),
+            AccessLevel::Exported => self.exported.as_ref(),
+            AccessLevel::Reachable => self.reachable.as_ref(),
+            AccessLevel::ReachableFromImplTrait => self.reachable_from_impl_trait.as_ref(),
+        }
```

Comment:
```suggestion
        }.as_ref()
```
`as_ref` can be written only once on the whole match.